packages/parameters/src/types/AppConfigProvider.ts

```diff
@@ -0,0 +1,50 @@
+import type {
+  AppConfigDataClientConfig,
+  StartConfigurationSessionCommandInput,
+} from '@aws-sdk/client-appconfigdata';
+import type { GetOptionsInterface } from 'types/BaseProvider';
+
+/**
+ * Options for the AppConfigProvider class constructor.
+ *
+ *  @interface AppConfigProviderOptions
+ *  @property {string} environment - The environment ID or the environment name.
+ *  @property {string} [application] - The application ID or the application name.
+ *  @property {AppConfigDataClientConfig} [clientConfig] - Optional configuration to pass during client initialization, e.g. AWS region.
+ */
+interface AppConfigProviderOptions {
+  environment: string
+  application?: string
+  clientConfig?: AppConfigDataClientConfig
+}
+
+/**
+ * Options for the AppConfigProvider get method.
+ *
+ * @interface AppConfigGetOptionsInterface
+ * @extends {GetOptionsInterface}
+ * @property {StartConfigurationSessionCommandInput} [sdkOptions] - Required options to start configuration session.
+ */
+interface AppConfigGetOptionsInterface extends Omit<GetOptionsInterface, 'sdkOptions'> {
+  sdkOptions?: Omit<
+  Partial<StartConfigurationSessionCommandInput>,
+  | 'ApplicationIdentifier'
+  | 'EnvironmentIdentifier | ConfigurationProfileIdentifier'
+  >
+}
+
+/**
+ * Combined options for the getAppConfig utility function.
+ *
+ * @interface getAppConfigCombinedInterface
+ * @extends {AppConfigProviderOptions, AppConfigGetOptionsInterface}
+ */
+interface getAppConfigCombinedInterface
+  extends AppConfigProviderOptions,
```

Comment:
Copying excerpt from [this other comment](https://github.com/awslabs/aws-lambda-powertools-typescript/pull/1200#issuecomment-1373569312).

My main question mark is on whether we should restrict the type to avoid passing `clientConfig` to the utility function, i.e.:
```ts
interface getAppConfigCombinedInterface
   extends Omit<AppConfigProviderOptions, 'clientConfig>,
   AppConfigGetOptionsInterface {}
```

The reason I suggest this is for coherence with the other providers. None of the other providers allows to customize the AWS SDK when using the utility functions. That should be a behavior (and a reason to use) for the class providers.

Reply:
Also can we capitalize the name to `GetAppConfigCombinedInterface` for uniformity?

Reply:
> The reason I suggest this is for coherence with the other providers. None of the other providers allows to customize the AWS SDK when using the utility functions. That should be a behavior (and a reason to use) for the class providers.

100% agree. Consistency is important, I'll make the changes.